crates/floresta-wire/src/p2p_wire/address_man.rs

```diff
@@ -389,6 +389,24 @@ impl AddressMan {
         Ok(())
     }
 
+    /// Dumps the good utreexo peers to a file on dir `datadir/anchors.json` in json format `
+    pub fn dump_utreexo_peers(&self, datadir: &str) -> std::io::Result<()> {
```

Comment:
Thanks for the review! I'll be working on it.

Reply:
If I run `self.peer_by_service.get(&service_flags::UTREEXO.into())` on `node.rs` it will return the connected peers filtered by service, right? 

In my understanding, the problem with my first take on this was that I retrieved the peers at `address_manager` level, instead of retrieving the peers at the running node. 

Therefore, this would be the modification on `node.rs`:
```

impl<T, Chain> UtreexoNode<T, Chain>
where
    T: 'static + Default + NodeContext,
    WireError: From<<Chain as BlockchainInterface>::Error>,
    Chain: BlockchainInterface + UpdatableChainstate + 'static,
{
...
    /// Saves the utreexo peers to disk so we can reconnect with them later
    pub(crate) fn save_utreexo_peers(&self) -> Result<(), WireError> {
        let utreexo_peers_connected = self
            .peer_by_service
            .get(&service_flags::UTREEXO.into())
            .ok_or(WireError::NoPeersAvailable)?;
        let utreexo_peers_connected = utreexo_peers_connected.iter().map(|peer| *peer as usize ).collect();
        self.address_man
            .dump_utreexo_peers(&self.datadir, utreexo_peers_connected)
            .map_err(WireError::Io)
    }
...
}
```
    
   The method now would retrieve the connected utreexo peers to the running node and dump them do the `anchors.json` file. 
   
   I had to modify the type of `peer` to `usize` due to the `.get()` method inside `.dump_utreexo_peers()` taking `&usize,` instead of `&u32` as argument.

Reply:
I'll try here. Any tips on how to make sure I am saving the connected utreexo peers and not any other utreexo peers?

Reply:
> If I run self.peer_by_service.get(&service_flags::UTREEXO.into()) on node.rs it will return the connected peers filtered by service, right?

Yes!

> In my understanding, the problem with my first take on this was that I retrieved the peers at address_manager level, instead of retrieving the peers at the running node.

Exactly!